### PR TITLE
Introduce Threaded Executions to the Runtime Server Instrument - Part 1

### PR DESCRIPTION
**Pull request by lolczak**
### Pull Request Description
Refactoring toward converting requests into commands and separation of a request for a
service from its execution.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/docs/style-guide/scala.md) and [Java](https://github.com/luna/enso/blob/master/docs/style-guide/java.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/execution/EnsoCommandProcessor.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  )
+
+  /** @inheritdoc **/
+  def invoke(cmd: Command, ctx: RuntimeContext): Future[Done.type] = {
+    val promise = Promise[Done.type]()
+    executor.submit[Unit](new Callable[Unit] {
+      override def call(): Unit = {
+        val logger = ctx.executionService.getLogger
+        logger.log(Level.FINE, s"Starting $cmd...")
+        cmd.execute(ctx)
```
Comment (4e6) [inline engine/runtime/src/main/scala/org/enso/interpreter/instrument/execution/EnsoCommandProcessor.scala:34]: It seems that if the execution fails, the promise will never be completed

Comment (lolczak) [reply engine/runtime/src/main/scala/org/enso/interpreter/instrument/execution/EnsoCommandProcessor.scala]: good point, fixed

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/EnsoExecutionSupport.scala [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+import org.enso.polyglot.runtime.Runtime.Api
+import org.enso.polyglot.runtime.Runtime.Api.ContextId
+
+import scala.jdk.javaapi.OptionConverters
+
+/**
+  * Provides support for executing Enso code. Adds convenient methods to
+  * run Enso programs in a Truffle context.
+  */
+trait EnsoExecutionSupport {
```
Comment (kustosz) [inline engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/EnsoExecutionSupport.scala:27]: Do not hardcode `Enso` in class and method names.

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/EnsoExecutionSupport.scala [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+  /**
+    * Runs an Enso program.
+    *
+    * @param contextId an identifier of an execution context
+    * @param stack a call stack
+    * @param ctx a runtime context
+    * @return either an error message or Unit signaling completion of a program
+    */
+  def runEnso(
```
Comment (kustosz) [inline engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/EnsoExecutionSupport.scala:103]: Same here, don't hardcode `enso`. Call it `runProgram` or something.

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala [hunk 4]
```diff
@@ -160,7 +160,11 @@ class MainModule(serverConfig: LanguageServerConfig) {
   )
 
   lazy val jsonRpcServer =
-    new JsonRpcServer(JsonRpc.protocol, jsonRpcControllerFactory)
+    new JsonRpcServer(
+      JsonRpc.protocol,
+      jsonRpcControllerFactory,
+      JsonRpcServer.Config(10000, 10.seconds)
```
[the remark this answers is not in the document]
Comment (lolczak) [reply engine/language-server/src/main/scala/org/enso/languageserver/boot/MainModule.scala]: done

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/EnsoExecutionSupport.scala (hunk 2 above)
Comment (lolczak) [reply engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/EnsoExecutionSupport.scala]: done

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/EnsoExecutionSupport.scala (hunk 3 above)
Comment (lolczak) [reply engine/runtime/src/main/scala/org/enso/interpreter/instrument/command/EnsoExecutionSupport.scala]: done

**Pull request merged**